sdk/Lusid.Sdk.Tests/Utilities/PollyApiRetryHandlerTests.cs

```diff
@@ -0,0 +1,409 @@
+﻿using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Lusid.Sdk.Api;
+using Lusid.Sdk.Client;
+using Lusid.Sdk.Utilities;
+using NUnit.Framework;
+using Polly;
+using RestSharp;
+
+namespace Lusid.Sdk.Tests.Utilities
+{
+    [TestFixture]
+    // For further Polly functionality reference, check out this page: https://github.com/App-vNext/Polly
+    public class PollyApiRetryHandlerTests
+    {
+        private LusidApiFactory _apiFactory;
+        private readonly Policy<IRestResponse> _initialRetryPolicy = RetryConfiguration.RetryPolicy;
+        private HttpListener _httpListener;
+        private const string ListenerUriPrefix = "http://localhost:4444/";
+
+        [SetUp]
+        public void SetUp()
+        {
+            _httpListener = new HttpListener();
+            _httpListener.Prefixes.Add(ListenerUriPrefix);
+
+            var testApiConfig = TestLusidApiFactoryBuilder.CreateApiConfiguration(null);
+            testApiConfig.ApiUrl = ListenerUriPrefix;
+
+            _apiFactory = new LusidApiFactory(testApiConfig);
+
+            _httpListener.Start();
+        }
+
+        [Test]
+        public void CallGetPortfoliosApi_WhenHttpStatusIs400AndRetryConditionIsNotSatisfied_ThrowsApiExceptionWithoutRetry()
+        {
+            // It should not retry on codes >400 by default, unless specified
+            const int expectedStatusCode = 400;
+            // Add the next response returned by api
+            AddMockHttpResponseToQueue(_httpListener, statusCode: expectedStatusCode, responseContent: "{\"some\": \"JsonResponseHere\"}");
+            var retryCount = 0;
+            RetryConfiguration.RetryPolicy =
+                Policy.Wrap(
+                    PollyApiRetryHandler.DefaultFallbackPolicy,
+                    Policy
+                        .HandleResult<IRestResponse>(PollyApiRetryHandler.GetInternalExceptionRetryCondition)
+                        .Retry(retryCount: 3, onRetry: (result, i) => retryCount++)
+                );
+        
+            var exception = Assert.Throws<ApiException>(
+                () => _apiFactory.Api<IPortfoliosApi>().GetPortfolio("any", "any")
+            );
+
+            Assert.That(exception.ErrorContent, Is.EqualTo("{\"some\": \"JsonResponseHere\"}"));
+            Assert.That(exception.ErrorCode, Is.EqualTo(expectedStatusCode));
+            Assert.That(retryCount, Is.EqualTo(0));
+        }
+
+        [Test]
+        public void CallGetPortfoliosApi_WhenHttpStatusIs200AndRetryConditionIsNotSatisfied_NoPollyRetryIsTriggered()
+        {
+            // It should do nothing when response code is 200
+            const int expectedStatusCode = 200;
+            // Add the next response returned by api
+            AddMockHttpResponseToQueue(_httpListener, expectedStatusCode, responseContent: "{\"some\": \"JsonResponseHere\"}");
```

Comment:
Why don't you just do `new Portfolio(...).ToJson()`?

Reply:
That's a good shout actually!